Added scroll progress indicator bar

meetarora10 opened the pull request:
# Fixes Issue🛠️
<!-- Example: Closes #32 -->

Closes #1597 

# Description👨‍💻

I have added a progress bar on the top of the navbar which progresses as the page scrolls up or down.

# Type of Change📄
<!-- Please delete the options that are not relevant to you. -->

- [x] New feature (non-breaking change which adds functionality)

# Checklist✅
<!-- Please delete the options that are not relevant to you. -->

- [x] I am an Open Source contributor
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas

# Screenshots/GIF📷
<!-- Please add screenshots or a GIF to demonstrate your changes. -->

https://github.com/user-attachments/assets/5e3ee77e-213e-40d2-8b8b-3c3eddd9b921

Rakesh9100 commented:
Please add the same in all the pages such as About, Contributors, etc.

Also, use the color that matches the home page UI nicely @meetarora10

meetarora10 commented:
Sure @Rakesh9100

meetarora10 commented:
@Rakesh9100 now I have added in all the pages and also changed the UI as you said .
![Screenshot (961)](https://github.com/user-attachments/assets/9892d30c-b166-447b-b0b0-0257ec86331e)
![Screenshot (962)](https://github.com/user-attachments/assets/d3dc08c4-b0ea-4c8b-a4bc-0f06e09f12a5)
![Screenshot (963)](https://github.com/user-attachments/assets/d6492855-1ab2-46bb-91b9-d965720397ae)
![Screenshot (964)](https://github.com/user-attachments/assets/8ab2f3b2-b46e-409d-a4b1-087ad17068e6)
![Screenshot (965)](https://github.com/user-attachments/assets/f094ea0e-3199-4078-945a-f1e690003f7a)
